File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let ns=s.trim().split(' ');
    let result = '';
    for (var i = ns.length - 1; i >= 0; i--) {
        if (ns[i] !== '') {
            result += `${ns[i]} `;
        }
    }
    return result.trim();
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] res=s.split("\s+");
    StringBuilder sb=new StringBuilder();
    for(int i=res.length-1;i>=0;i--){
        sb.append(res[i].trim()+" ");
    }
    return sb.toString().trim();
    }
}